src/services/utilities.ts

```diff
@@ -436,6 +436,35 @@ namespace ts {
         return isInCommentHelper(sourceFile, position, /*predicate*/ undefined);
     }
 
+    export function isInJsxAttribute(sourceFile: SourceFile, position: number) {
+        let token = getTokenAtPosition(sourceFile, position);
+        if (token && (token.kind === SyntaxKind.JsxAttribute || token.kind === SyntaxKind.JsxSpreadAttribute)) {
+            return true;
+        }
+
+        if (token && token.kind === SyntaxKind.CloseBraceToken && token.parent.kind === SyntaxKind.JsxExpression) {
+            return true;
+        }
+
+        return false;
+    }
+
+    export function isInJsxElement(sourceFile: SourceFile, position: number) {

```

Comment:
What do you mean *in*? Inside the opening tag? The closing tag? The children? Does this include `JsxSelfClosingElement`s?